streams/src/test/java/org/apache/kafka/streams/kstream/KStreamBuilderTest.java

```diff
@@ -79,13 +79,48 @@ public void testNewName() {
         assertEquals("Y-0000000001", builder.newName("Y-"));
         assertEquals("Z-0000000002", builder.newName("Z-"));
 
-        KStreamBuilder newBuilder = new KStreamBuilder();
+        final KStreamBuilder newBuilder = new KStreamBuilder();
 
         assertEquals("X-0000000000", newBuilder.newName("X-"));
         assertEquals("Y-0000000001", newBuilder.newName("Y-"));
         assertEquals("Z-0000000002", newBuilder.newName("Z-"));
     }
 
+
+    @Test
+    public void shouldHaveSinkTopicForStreamTo() {
```

Comment:
nit: The name of the test is a bit misleading, since we are actually not trying to fetch any data from `topic-sink` at all, but just making sure that it will not throw a NPE in this function, and the checking is actually to make sure the record from `topic-source` has indeed piped to the processor after it, which is covered in other test coverages already.

So I felt we can get rid of the check at line 121, and rename it to `shouldNotTryProcessingFromSinkTopic`? And similarly the below test could be `shouldTryProcessingFromThoughTopic`?